frontend/src/components/StakeSelectorItem.vue

```diff
@@ -1,217 +1,874 @@
+      <multiselect v-if="isNftStaking" class="multiselect" v-model="idsToStake"
+        :options="(isDeposit ? ownedLandIds : stakedIds).filter(x => +x.tier === +tier || !tier) || []"
+        :max="10" :multiple="true" :close-on-select="false" :clear-on-select="false" :placeholder="$t('stake.pickId')"
+        :custom-label="idWithTier" track-by="id"
+        tagPosition="bottom">
+        <template slot="selection" :slot-scope="{ idsToStake }">
+          <span class="multiselect__single" v-if="idsToStake && idsToStake.length">{{ idsToStake.length }} {{$t('stake.idsSelected')}}</span>
+        </template>
+        <template slot="noOptions">
+          <span>{{(isDeposit ? $t('stake.StakeSelectorItem.noLandStaked') : $t('stake.StakeSelectorItem.noLandToStake'))}}</span>
+        </template>
+      </multiselect>
+      <button class="stake-button stake-submit-button"
+        @click="onSubmit"
+        :disabled="currentState !== CurrentState.ok">
+        <span v-if="isLoadingStake"><i class="fa fa-spinner fa-spin"></i></span>
+        <span v-else>{{ submitButtonLabel }}</span>
+        </button>
+    </div>
   </div>
 </template>
 
-<script>
+<script lang="ts">
+import Vue from 'vue';
 import { formatDurationFromSeconds } from '../utils/date-time';
-import { isNftStakeType } from '../interfaces/State';
-
-export default {
-  props: [
-    'stakeTitle',
-    'stakeTokenName',
-    'rewardTokenName',
-    'stakeType',
-    'minimumStakeTime',
-    'estimatedYield',
-    'rewardsDuration',
-    'deprecated',
-    'rewardDistributionTimeLeft',
-    'currentRewardEarned'
-  ],
+import { isStakeType, isNftStakeType, StakeType, NftStakeType } from '../interfaces/State';
+import { toBN } from '@/utils/common';
+import { secondsToDDHHMMSS } from '../utils/date-time';
+import {Multiselect} from 'vue-multiselect';
+import {PropType} from 'vue/types/options';
+import BN from 'bignumber.js';
+import {mapActions, mapState} from 'vuex';
+import i18n from '@/i18n';
+import { TranslateResult } from 'vue-i18n';
+
+interface StoreMappedActions {
+  fetchStakeDetails(payload: {stakeType: StakeType | NftStakeType}):
+  Promise<void>;
+  stake(payload: {amount: string; stakeType: StakeType | NftStakeType}):
+  void;
+  stakeNfts(payload: {ids: string[]; stakeType: StakeType | NftStakeType}):
+  void;
+  unstake(payload: {amount: string; stakeType: StakeType | NftStakeType}):
+  void;
+  unstakeNfts(payload: {ids: string[]; stakeType: StakeType | NftStakeType}):
+  void;
+  unstakeKing(payload: {amount: string;}):
+  Promise<void>;
+  claimKingReward():
+  Promise<void>;
+  stakeUnclaimedRewards(payload: {stakeType: StakeType;}):
+  Promise<void>;
+  claimReward(payload: {stakeType: StakeType | NftStakeType;}):
+  Promise<void>;
+  getOwnedLandIdsWithTier():
+  Promise<LandIds[]>;
+  getStakedIds(payload: {stakeType: StakeType | NftStakeType;}):
+  Promise<string[]>;
+}
+
+enum CurrentState {
+  connectWallet,
```

Comment:
Generally enum values should be capitalized, like, connectWallet -> CONNECT_WALLET

Reply:
done [9115fbe](https://github.com/CryptoBlades/cryptoblades/pull/1287/commits/9115fbe2306749216362df1ebece5b1d75a76b2b)